extensions/exquery/restxq/src/main/java/org/exist/extensions/exquery/restxq/impl/RestXqServiceImpl.java

```diff
@@ -297,4 +319,185 @@ private static StringValue parseAsString(final InputStream is, final String enco
         final String s = new String(bos.toByteArray(), encoding);
         return new StringValue(s);
     }
+    
+    /**
+     * Extract parts from multipart/related request and build XML http:multipart Node as String including body parts and headers
+     * The first body element is the root part.
+	 * Body content is represented as BASE64
+	 * 
+     * <http:multipart>
+  	 * 		<http:header name="content-id" value="..." />
+  	 *		<http:body media-type="...">...</http:body>
+	 *		.
+	 *		. 
+	 * 	</http:multipart>
+	 * 
+     */
+    private String extractBodyParts(final HttpRequest request, final InputStream is, final String encoding) throws RestXqServiceException, IOException{
+    	
+    	final String requestContentType = request.getContentType();
+    	final String boundary = getMultiPartBoundary(requestContentType);
```

Comment:
I remember that the spec allows multiple boundary definitions in one request.
It is ok for me not to support this, but we must be aware.....

Reply:
That's true, in case of nested multipart entities "a Content-Type of multipart in a body part within another multipart entity": [Multipart RFC 1341](https://www.w3.org/Protocols/rfc1341/7_2_Multipart.html)
In this case further implementation is needed and a TODO task at the moment